core-plugin/src/com/google/gct/idea/debugger/CloudDebuggerRunner.java

```diff
@@ -74,48 +87,48 @@ public String getRunnerId() {
   }
 
   private static RunContentDescriptor createContentDescriptor(@Nullable final CloudDebugProcessState processState,
-                                                              @NotNull final ExecutionEnvironment environment)
-    throws ExecutionException {
-    final XDebugSession debugSession =
-      XDebuggerManager.getInstance(environment.getProject()).startSession(environment, new XDebugProcessStarter() {
-        @NotNull
-        @Override
-        public XDebugProcess start(@NotNull final XDebugSession session) throws ExecutionException {
-
-          // Clear out the stash state which is queried on debug exit.
-          if (processState != null) {
-            ProjectRepositoryState.fromProcessState(processState).clearForNextSession();
-          }
-
-          CloudDebugProcessState state = processState;
-          CloudAttachDialog attachDialog = new CloudAttachDialog(session.getProject());
-          attachDialog.setInputState(state);
-          DialogManager.show(attachDialog);
-          state = attachDialog.getResultState();
-
-          ProjectRepositoryValidator validator = null;
-          if (state != null) {
-            validator = new ProjectRepositoryValidator(state);
-          }
-          if (!attachDialog.isOK() || state == null || !validator.isValidDebuggee()) {
-            throw new RunCanceledByUserException();
-          }
+      @NotNull final ExecutionEnvironment environment) throws ExecutionException {
 
-          if (environment.getRunnerAndConfigurationSettings() != null &&
-              environment.getRunnerAndConfigurationSettings()
-                .getConfiguration() instanceof CloudDebugRunConfiguration) {
-            CloudDebugRunConfiguration config =
-              (CloudDebugRunConfiguration)environment.getRunnerAndConfigurationSettings().getConfiguration();
-            // State is only stored in the run config between active sessions.
-            // Otherwise, the background watcher may hit a check during debug session startup.
-            config.setProcessState(null);
+    final XDebugSession debugSession =
+        XDebuggerManager.getInstance(environment.getProject()).startSession(environment, new XDebugProcessStarter() {
+          @NotNull
+          @Override
+          public XDebugProcess start(@NotNull final XDebugSession session) throws ExecutionException {
+
+            // Clear out the stash state which is queried on debug exit.
+            if (processState != null) {
+              ProjectRepositoryState.fromProcessState(processState).clearForNextSession();
+            }
+
+            CloudDebugProcessState state = processState;
+            CloudAttachDialog attachDialog = new CloudAttachDialog(session.getProject());
+            attachDialog.setInputState(state);
+            DialogManager.show(attachDialog);
+            state = attachDialog.getResultState();
```

Comment:
can we avoid reusing the variables? Maybe declare it here and just pass processState in line 105?